### PR TITLE
adding support to java filetype

### PR DESCRIPTION
**Pull request by xavychan**
Hi, I just added the support for outlining the methods in Java file. Kindly review and make a test with any .java file.

Please let me know your feedback.

Thanks,
Xavier Francis.

### CODE REVIEW

File: ftplugin/java.vim [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    var tmp_string = "<hshnnTejwqik93la,AMK3N2MNMAKPD+03mn2nhkalpdpk3nsla>"
+    for item in outline
+        i = i + 1
+    endfor
+    return outline
+enddef
+
+import autoload "../lib/ftfunctions/java.vim"
+
+b:OutlinePreProcessInternal = OutlinePreProcessInternal
```
Comment (ubaldot) [inline ftplugin/java.vim]: The `OutlinePreProcessInternal( ... )` function could be safely removed along with `b:OutlinePreProcessInternal = OutlinePreProcessInternal. `

File: lib/ftfunctions/java.vim [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import autoload "../regex.vim"
+
+export def CurrentItem(curr_item: string): string
+    #return trim(matchstr(curr_item, '(.*'))
+    return curr_item
+enddef
+
+export def FilterOutline(outline: list<string>): list<string>
+    outline ->filter("v:val =~ " .. string(join(regex.outline_pattern_to_include["java"], '\|')))
+    outline ->filter("v:val =~ " .. string(join(regex.outline_pattern_to_exclude["java"], '\|')))
```
Comment (ubaldot) [inline lib/ftfunctions/java.vim]:
```diff
-    outline ->filter("v:val =~ " .. string(join(regex.outline_pattern_to_exclude["java"], '\|')))
+            ->filter("v:val =~ " .. string(join(regex.outline_pattern_to_exclude["java"], '\|')))
```

Comment (ubaldot) [reply lib/ftfunctions/java.vim]: You can chain filters.

**Review by ubaldot — CHANGES_REQUESTED** [review]
Thanks! I added few comments!

File: ftplugin/java.vim (hunk 1 above)
Comment (xavychan) [reply ftplugin/java.vim]: applied this suggestion in latest commit, please review

File: lib/ftfunctions/java.vim (hunk 2 above)
Comment (xavychan) [reply lib/ftfunctions/java.vim]: applied the suggestion in latest commit, please review

**Comment by xavychan** [comment]
Hi, thanks for the review, applied the suggestions in the latest commit, kindly feedback.

**Review by ubaldot — APPROVED** [review]
It looks good now. Thanks!

**Pull request merged**